src/client/app/effects/github.ts

```diff
@@ -98,14 +103,20 @@ export class GitHubEffects {
             description.replace(Strings.gistDescriptionAppendage, ''); // shouldn't be necessary
             description += Strings.gistDescriptionAppendage;
 
+            if (type === GitHub.GitHubActionTypes.UPDATE_GIST) {
+                gistId = gist;
+            }
+
             return this._github.createOrUpdateGist(
                 `${description}`,
                 files,
-                null,
-                type === GitHub.GitHubActionTypes.SHARE_PUBLIC_GIST
+                gistId,
+                type === GitHub.GitHubActionTypes.SHARE_PUBLIC_GIST).map((gist: IGist) => {
```

Comment:
could we make this whole thing read cleaner?

    let isPublicGist = type === GitHub.GitHubActionTypes.SHARE_PUBLIC_GIST
    return this._github
        .createOrUpdateGist(
            description, files, gistId, isPublicGist
        )
        .map(gist: IGist => ({ gist: gist, snippetId: id} ));